library/Booster/Pattern/ApplyEquations.hs

```diff
@@ -318,34 +322,42 @@ runEquationT doTracing definition llvmApi smtSolver sCache (EquationT m) = do
     (res, endState) <-
         flip runStateT (startState sCache) $
             runExceptT $
-                runReaderT m EquationConfig{definition, llvmApi, smtSolver, doTracing}
+                runReaderT
+                    m
+                    EquationConfig
+                        { definition
+                        , llvmApi
+                        , smtSolver
+                        , doTracing
+                        , maxIterations = UnsafeGlobalState.unsafeReadGlobalMaxIterations
+                        , maxRecursion = UnsafeGlobalState.unsafeReadGlobalMaxRecursion
```

Comment:
We are actually in `IO` here so we could make the `UnsafeGlobalState` slighly less unsafe (still using a hidden global variable but no `unsafePerformIO`)

Reply:
Done.